### PR TITLE
Add basic data_processing test

### PR DESCRIPTION
**Pull request by juberti**
Trying to increase our test coverage...

### CODE REVIEW

File: ultravox/model/data_processing_test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import numpy as np
+import pytest
+import torch
+
+from ultravox.data import datasets
+from ultravox.model import data_processing
+
+TEST_USER_MESSAGE = {
+    "role": "user",
+    "content": "Listen to <|audio| and respond.",
```
Comment (zqhuang211) [inline ultravox/model/data_processing_test.py:12]: `<|audio|` should be `<|audio|>`

File: ultravox/model/data_processing_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+def test_process_inference_mode(mock_processor, fake_dataset):
+    dataproc = data_processing.UltravoxDataproc(
+        fake_dataset, mock_processor, inference_mode=True
+    )
+    processed = next(iter(dataproc))
+
+    assert processed["input_ids"].shape == torch.Size([5])
+    assert processed["attention_mask"].shape == torch.Size([5])
+    assert processed["labels"].equal(torch.tensor([0, 1, 2, 3, 4]))
```
Comment (farzadab) [inline ultravox/model/data_processing_test.py:79]: nit: we probably don't care about `labels` in inference_mode.

**Review by farzadab — APPROVED** [review]
LGTM
